app.go

```diff
@@ -0,0 +1,55 @@
+package httpcache
+
+import (
+	"errors"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/darkweak/souin/cache/surrogate/providers"
+	"github.com/darkweak/souin/cache/types"
+	"github.com/darkweak/souin/configurationtypes"
+)
+
+// SouinApp contains the whole Souin necessary items
+type SouinApp struct {
+	*DefaultCache
+	Provider         types.AbstractProviderInterface
+	SurrogateStorage providers.SurrogateInterface
+	API              configurationtypes.API `json:"api,omitempty"`
+	LogLevel         string                 `json:"log_level,omitempty"`
+}
+
+func init() {
+	caddy.RegisterModule(SouinApp{})
+}
+
+// Provision implements caddy.Provisioner
+func (s *SouinApp) Provision(_ caddy.Context) error {
+	return nil
+}
+
+// Start will start the App
+func (s SouinApp) Start() error {
+	if s.DefaultCache != nil && s.DefaultCache.GetTTL() == 0 {
```

Comment:
(Not sure without a deeper look, but) it's possible that these fields aren't ready until the app is started. I know the `tls` app doesn't get to verify some things until after it's started since other apps may modify its config at Provision-time.

But yeah, normally most modules would do verification in the Validate() step.

Reply:
The `Validate()` function is executed before the module provisioning and this make a nil DefaultCache property if we can this into the validate.

Reply:
Validate() is executed _after_ Provision(), but at the same "phase" of the config load with provisioning.

Reply:
But if I try to check through the `Validate` method inside the App context, DefaultCache is not already populated from the runners.

Reply:
I think the idea is that `Provision()` should be filling `s.DefaultCache`.

Reply:
Oh, right, Validate() may be called before _other_ modules' Provision() is called, but is always called after the same modules' Provision().